### PR TITLE
Fix TF WarmUp class

### LINKED ISSUE #5098: 🐛 [TF] `create_optimizer` wrong superposition of learning rate schedules
# 🐛 Bug

When using `create_optimizer`, 2 learning rate schedulers are placed on top of each other (WarmUp and keras Polynomial Decay) :

https://github.com/huggingface/transformers/blob/efeb75b8054cc299698cf8bc09f395ada2660745/src/transformers/optimization_tf.py#L70-L80

But the step in Warmup scheduler is not updated to take into account the warmup steps, which lead to a wrong learning rate :

image

---

Expected learning rate shape :

Dessin sans titre

### PR DESCRIPTION
**Pull request by astariul**
This commit fixes the TF WarmUp learning rate scheduler.

The LR shape was wrong due to warmup steps. See linked issue for more details.

Fix #5098

### CODE REVIEW

**Comment by jplu** [comment]
Thanks @Colanim! Good catch!! Unfortunately it is a duplicate of #4940 :smile: and should be merged soon :)

**Comment by Colanim** [comment]
Haha didn't see x)
